app/helpers/hyku_addons/cross_tenant_shared_search_helper.rb

```diff
@@ -4,28 +4,23 @@
 module HykuAddons
   module CrossTenantSharedSearchHelper
     def generate_work_url(model, request)
+      # returns a symbol for id and string for other attributes
       model = model.with_indifferent_access
-      request_protocol = request.protocol || 'http://'
-      request_port = request.port
-      request_host = request.host
 
       cname = model["account_cname_tesim"]
-      account_cname = cname.class == Array ? cname.try(:first) : cname
-
+      account_cname = Array.wrap(cname).first
       has_model = model["has_model_ssim"].first.underscore.pluralize
-      # returns a symbol not a string
       id = model[:id]
-      request_params = { request_protocol: request_protocol, request_host: request_host, request_port: request_port }
 
-      if has_model == "collections"
-        set_collection_url(id: id, request: request_params, account_cname: account_cname, has_model: has_model)
-      else
-        set_work_url(id: id, request: request_params, account_cname: account_cname, has_model: has_model)
-      end
+      request_params = %i[protocol host port].map { |method| ["request_#{method}".to_sym, request.send(method)] }.to_h
+      set_url(id: id, request: request_params, account_cname: account_cname, has_model: has_model)
```

Comment:
This is now the only place where `id` is being referenced, so we could replace this with `model["id"]` and save the variable assignment.

Reply:
I can remove the varibale assignment but  **model[:id]** will stay as it is,  because I discovered the hash from solr returns **id** as symbol and the other attributes we are using here eg **has_model_ssim** as string.